### DISCUSSION TITLE
Does operands order matter for spectral distribution arithmetical operations?

### DISCUSSION THREAD

**Discussion by cuixiaofei** [colour-science/colour#1071] [Q&A]
### Question

Hello, I have seen the arithmetical operations available in tutorials.
But today, it can't work well, could you help?

```python
Traceback (most recent call last):
  File "MyColormix.py", line 172, in <module>
    sd_result = (16 * red_sd) + (10 * green_sd) + (12 * blue_sd)
TypeError: unsupported operand type(s) for *: 'int' and 'SpectralDistribution'
```
float number also can not multiply with SpectralDistribution

**Comment by thomasmansencal** [comment]
Hi @cuixiaofei,

This is a great question! We define various under methods, e.g. `colour.SpectralDistribution.__add__` and this is what makes the various arithmetical operations possible, e.g. `my_sd + 1`. For the builtin type `int` however, there is no `__add__` method that support a `colour.SpectralDistribution` instance so a `TypeError` is raised.

Try that instead:

```python
>>> import colour
>>> red_sd = colour.sd_ones()
>>> blue_sd = colour.sd_ones()
>>> sum_sd = (red_sd * 16) + (blue_sd * 10)
```

Cheers,

Thomas
